source/Xunit.Gherkin.Quick/CoreModel/ScenarioOutlineExecutor.cs

```diff
@@ -34,10 +34,12 @@ public ScenarioOutlineExecutor(IFeatureFileRepository featureFileRepository)
             if (gherkinScenarioOutline == null)
                 throw new InvalidOperationException($"Cannot find scenario outline `{scenarioOutlineName}`.");
 
-            var gherkinScenario = gherkinScenarioOutline.ApplyExampleRow(exampleName, exampleRowIndex);
-			var gherkinBackground = featureFile.GetBackground();
+            var gherkinBackground = featureFile.GetBackground();
+            var gherkinScenario = gherkinScenarioOutline
+                .ApplyExampleRow(exampleName, exampleRowIndex)
+                .ApplyBackground(gherkinBackground);
```

Comment:
Same here. This call (ApplyBackhground) should only happen if the scenario is not null.
BTW, to be clear, here you are calling ApplyBackground on the scenario, not on ScenarioOutline, because ApplyExampleRow returns Scenario and not ScenarioOutline.
Which is perfect - exactly what I had in mind.
So the other extension method which you defined for the scenarioOutline is not really needed. That answers one of my comments. We don't need the extension method on ScenarioOutline.